### PR TITLE
[CARBONDATA-1611][Streaming] Reject Update and Delete operation for streaming table

### PR DESCRIPTION
**Pull request by jackylk**
Streaming table is not mutable, data update and delete operation should be rejected.
An exception will be thrown in the scenario.

 - [X] Any interfaces changed?
    No

 - [X] Any backward compatibility impacted?
    No

 - [X] Document update required?
    Yes

 - [X] Testing done
   Yes, one testcase added
      
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
    NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/730/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1367/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/733/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1370/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/744/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1381/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    intercept[MalformedCarbonCommandException] {
+      sql("""DELETE FROM source WHERE d.c1 = 'a'""").show()
+    }
+  }
+
+  override def afterAll {
+    sql("USE default")
+    sql("DROP DATABASE IF EXISTS streaming CASCADE")
+    CarbonProperties.getInstance()
```
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: These code no need to add.
-------------------------
 CarbonProperties.getInstance()
    .addProperty(CarbonCommonConstants.isHorizontalCompactionEnabled , "true")
   CarbonProperties.getInstance()
    .addProperty(CarbonCommonConstants.ENABLE_VECTOR_READER , "true")

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  test("test blocking update and delete operation on streaming table") {
+    intercept[MalformedCarbonCommandException] {
+      sql("""UPDATE source d SET (d.c2) = (d.c2 + 1) WHERE d.c1 = 'a'""").show()
+    }
+    intercept[MalformedCarbonCommandException] {
+      sql("""DELETE FROM source WHERE d.c1 = 'a'""").show()
+    }
+  }
+
+  override def afterAll {
```
Comment (chenliang613) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: in afterAll {}, please drop all created tables.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/StreamingTableStrategy.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.execution.{SparkPlan, SparkStrategy}
+import org.apache.spark.sql.execution.command.ExecutedCommandExec
+import org.apache.spark.sql.execution.command.mutation.{DeleteExecution, ProjectForDeleteCommand, ProjectForUpdateCommand}
+import org.apache.spark.sql.hive.CarbonRelation
+
+import org.apache.carbondata.spark.exception.MalformedCarbonCommandException
+
+/**
+ * Strategy for streaming table, like blocking unsupported operation
```
Comment (chenliang613) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/StreamingTableStrategy.scala]: align the comment.

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala (hunk 1 above)
Comment (jackylk) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: fixed

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala (hunk 2 above)
Comment (jackylk) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: DROP DATABASE IF EXISTS streaming CASCADE will drop all tables

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/StreamingTableStrategy.scala (hunk 3 above)
Comment (jackylk) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/StreamingTableStrategy.scala]: fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/782/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1415/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/797/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1431/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/815/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1447/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1449/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**